src/pages/settings/user/components/Details.tsx

```diff
@@ -33,36 +43,80 @@ export function Details() {
         <Card title={t('details')}>
           <Element leftSide={t('first_name')}>
             <InputField
-              id="first_name"
               value={userChanges?.first_name || user?.first_name || ''}
-              onChange={handleChange}
+              onValueChange={(value) =>
+                handleChange('first_name', value as string)
+              }
             />
           </Element>
 
           <Element leftSide={t('last_name')}>
             <InputField
-              id="last_name"
               value={userChanges?.last_name || user?.last_name || ''}
-              onChange={handleChange}
+              onValueChange={(value) =>
+                handleChange('last_name', value as string)
+              }
             />
           </Element>
 
           <Element leftSide={t('email')}>
             <InputField
-              id="email"
               value={userChanges?.email || user?.email || ''}
               type="email"
-              onChange={handleChange}
+              onValueChange={(value) => handleChange('email', value as string)}
             />
           </Element>
 
           <Element leftSide={t('phone')}>
             <InputField
-              id="phone"
               value={userChanges?.phone || user?.phone || ''}
-              onChange={handleChange}
+              onValueChange={(value) => handleChange('phone', value as string)}
             />
           </Element>
+
+          {company?.custom_fields?.user1 && (
+            <CustomField
+              field="user1"
+              defaultValue={userChanges.custom_value1}
+              value={company.custom_fields.user1}
+              onValueChange={(value) =>
+                handleChange('custom_value1', value as string)
+              }
+            />
+          )}
+
+          {company?.custom_fields?.user2 && (
+            <CustomField
+              field="user2"
+              defaultValue={userChanges.custom_value2}
+              value={company.custom_fields.user2}
+              onValueChange={(value) =>
+                handleChange('custom_value2', value as string)
```

Comment:
No need to typecast this `as string`, just modify signature of base function.

Reply:
removed